apps/api/src/goods-receipts/goods-receipts.controller.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  UsePipes,
+  ValidationPipe,
+  BadRequestException
+} from '@nestjs/common';
+import { GoodsReceiptsService } from './goods-receipts.service';
+import { CreateGoodsReceiptDto } from './dto/create-goods-receipt.dto';
+import { UpdateGoodsReceiptDto } from './dto/update-goods-receipt.dto';
+
+@Controller('goods-receipts')
+export class GoodsReceiptsController {
+  constructor(private readonly goodsReceiptsService: GoodsReceiptsService) {}
+
+  @Post()
+  @UsePipes(new ValidationPipe({ transform: true }))
+  create(@Body() createGoodsReceiptDto: CreateGoodsReceiptDto) {
+    return this.goodsReceiptsService.create(createGoodsReceiptDto);
+  }
+
+  @Get()
+  findAll() {
+    return this.goodsReceiptsService.findAll();
+  }
+
+  @Get(':id')
+  findOne(@Param('id') id: string) {
+    const numericId = Number(id);
+    if (Number.isNaN(numericId)) {
+      throw new BadRequestException('Invalid ID format');
+    }
+    return this.goodsReceiptsService.findOne(numericId);
+  }
+
+  @Patch(':id')
+  @UsePipes(new ValidationPipe({ transform: true }))
```

Comment:
There's a global pipe pattern that will do this app wide too..

Reply:
I saw that you could do this via the bootstrap function and at the time I didn’t know if there were unintended consequences to this so I went with explicitly applying it where needed. Now, as I dig more into it, seems like only request objects with validate or transform decorators will kick off the validation so it shouldn’t have unintended consequences. Am I thinking about this correctly?